src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingFurnaceUser.java

```diff
@@ -0,0 +1,98 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.coremod.colony.Colony;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockFurnace;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.Constants.*;
+import static com.minecolonies.api.util.constant.Suppression.MAGIC_NUMBERS_SHOULD_NOT_BE_USED;
+import static com.minecolonies.api.util.constant.Suppression.OVERRIDE_EQUALS;
+
+/**
+ * Abstract Class for all furnace users.
+ */
+@SuppressWarnings(OVERRIDE_EQUALS)
+public abstract class AbstractBuildingFurnaceUser extends AbstractBuildingWorker
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)